summingbird-storm/src/main/scala/com/twitter/summingbird/storm/collector/AggregatorOutputCollector.scala

```diff
@@ -16,50 +16,60 @@ import java.util.{ List => JList }
  *
  * AggregatorOutputCollector is a wrapper around the SpoutOutputCollector.
  * AsyncSummer is used to aggregate the tuples.
- * Different streams have seperated aggregators and caches.
- *
+ * Different streams have separate aggregators and caches.
  */
 class AggregatorOutputCollector[K, V: Semigroup](
     in: SpoutOutputCollector,
     summerBuilder: SummerBuilder,
-    summerShards: KeyValueShards) extends SpoutOutputCollector(in) {
+    summerShards: KeyValueShards,
+    flushExecTimeCounter: Incrementor,
+    executeTimeCounter: Incrementor) extends SpoutOutputCollector(in) {
 
-  // Map keeps track of summers corresponding to streams.
+  // An individual summer is created for each stream of data. This map keeps track of the stream and its corresponding summer.
   private val spoutCaches = MMap[String, AsyncSummer[(K, V), Map[K, V]]]()
 
-  // The Map keeps track of batch of aggregated tuples' messageIds. It also has a stream level tracking.
-  private val streamMessageIdTracker = MMap[String, MMap[Int, MList[Object]]]()
+  // As the crushDown happens at a stream level. We have a mapping from stream to the messageIds.
+  // The messageIds are further aggregated to the level of summerShards.
+  // The Map keeps track of aggregated tuples' messageIds as a list.
+  private val cachesByStreamId = MMap[String, MMap[Int, MList[Object]]]()
```

Comment:
we should probably use `LongMap` for the inner map and avoid the boxing that comes with using generics. There is no `IntMap` in scala, but `LongMap` should be fine for our purporses, I think.